### ISSUE TITLE
Wrapping Resolver w/ Graphback?

### ISSUE THREAD

**Issue by dijbi** [aerogear/graphback#2250]
Hi,

I would like  to encrypt a input variable in Query/Mutation/Subscription.

I have a GraphQL enpoint (which i don't own) with a list of nonEncryptedID and i would have in my graphback backend all theses IDs either hashed or/and encrypted.

Example :

//ENDPOINT i don't own //
```
Note {
  notEncryptedD  ; String!
}
 
// ENDPOINT I own
NoteInput {
  notEncryptedID: String!
}

AdditionalNoteData{
  id:ID! <-- either encrypted or hashed
  comments: [Comment]
}

type Comment {
  data: String
}
 
Query {
  getNoteByID(input:NoteInput) : AdditionalNoteData
}
```

With a Graphback Plugin, i am adding two fields to Note (hashofID & EncryptedID) 
Graphback is generated the CRUD starndard querys/resolvers/subscriptions, i would like to wrap theses Queries/Resolvers/Subscriptions to make the encryption/Hashing. 

Wrapping the Resolvers/Query/Mutation seemed a good idea (cf. [wrapResolvers] (https://graphql-compose.github.io/docs/basics/what-is-resolver.html#via-resolverwrapresolve) in graphql-compose).

I would love to filter the type having the encryption logic (in the fake example only Note but not Comment). My best option would be to add some kind of custom annotation to automate this logic but i dont think it's possible.

Thaks for your feedback

**Comment by wtrocki** [comment]
Sounds like XY problem. What we are trying to achieve with this hashing?

Graphback returns object containing resolvers so wrapping is trivial

**Comment by dijbi** [comment]
Thanks for the reply.

Haven't heard about the XY problem... i got some readings to do.

What i want to achieve with the hashing/encryption.
- **first endpoint** GraphQL express : personnal data including a userID (i can query but i don't own)
- **second endpoint** : Additional Data on User (where the userID has to be encrypted)
- **gateway** where the stitching takes place. 

Since i want to create as many endpoints with additional user data & encryption, my best bet is to :
- create the CRUD schema with Graphback.
- add the Cypher/Encryption plugin to the newly created schema with Resolvers/Query/Mutations.
 - add the newly endoint to the gateway without the gateway knowing about encryption/hashing.

Since  the queries would most likely be ;
```
User {
 userdatafield
  additionnalData {
    field1
    field2
  }
}

```
I thought that having a hash of the userID inside the graphback endpoint would decrease the request time. But i still want to have a encrypted userID inside the graphback endpoint to compute (possible but it doesn't matter if it is time consuming) the userID.

Since graphql-compose wrapResolver used a ObjectTC to do the wrapping, i was wondering if we could access the ObjectTC in the graphback component (graphbackmedata??)

**Comment by wtrocki** [comment]
There are two things here:

## General practice

Generally, passing userID from the client doesn't seem to be secure and good practice. Encryption itself is guaranteed on transport level - https. Mentioned hashing can be the way to validate if the requested userID, matches the session user ID (however this might be a redundant operation)

## Solution for schema modification

If there is any unknown motive to go with the encryption I think this could be done using Graphback plugins: https://graphback.dev/docs/plugins/create

Basically, in the plugin, we can alter schema as we want (and underlying resolvers as well).
In plugins, we can also wrap resolvers etc.

Alternative will be to wrap resolvers and schema in your application after they are returned from Graphback - this will work if we want this behavior for certain things that will require numerous if operations in the plugin itself (that tends to be generic)

**Comment by arnab15** [comment]
hello,
I have knowledge of javascript ,nodejs,reactjs,expressjs, Mongodb but I haven't contributed to any open-source project before so I'm a beginner. I would like to contribute to this project.
how should I start can anyone suggest to me the way should I start contributing.

**Comment by craicoverflow** [comment]
Hi @arnab15 - please use the [](https://github.com/aerogear/graphback/discussions) to ask this question as your question is unrelated to this issue.

**Comment by arnab15** [comment]
@craicoverflow thanks.

**Comment by dijbi** [comment]
@wtrocki Thanks a lot for the detailled answer.

To follow the wrapping issue, there is no Resolvers attached to an ObjectType (via ObjectTypeComposer (OTC) functions) per se in the graphback GraphbackCoreMetadata object.

Fetching for the ObjectType Resolvers with
`schemaComposer.getOTC(modelName),getResolvers()
.`
yields no attached resolvers.

Using [wrapping functions](https://graphql-compose.github.io/docs/basics/what-is-resolver.html#wrapping-resolver) of graphql-compose is therefore not possible.

It is by design ?

**Comment by wtrocki** [comment]
We encourage developers to use good patterns. Meaning that we do not provide executable schema that one cannot edit. We split resolvers and schema so developers have freedom to wrap things. 

If you looking into building plugin then our plugins have dedicated place for dealing with resolvers:
https://graphback.dev/docs/plugins/create#createresolvers

There is no need to use schemaComposer apart from actual schema changes.

**Comment by dijbi** [comment]
Ok  Thanks a lot. The graphback plugin system seems adequate for various use cases .
